opentasks/src/main/java/org/dmfs/tasks/share/ShareTaskText.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright 2016 Marten Gajda <marten@dmfs.org>
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.tasks.share;
+
+import android.content.Context;
+import android.support.annotation.StringRes;
+import android.text.TextUtils;
+import android.text.format.Time;
+
+import org.dmfs.provider.tasks.TaskContract;
+import org.dmfs.tasks.R;
+import org.dmfs.tasks.model.CheckListItem;
+import org.dmfs.tasks.model.ContentSet;
+import org.dmfs.tasks.model.Model;
+import org.dmfs.tasks.model.TaskFieldAdapters;
+import org.dmfs.tasks.model.adapters.TimeZoneWrapper;
+import org.dmfs.tasks.utils.AbstractStringCharSequence;
+import org.dmfs.tasks.utils.DateFormatter;
+import org.dmfs.tasks.utils.TaskText;
+
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+import java.util.TimeZone;
+
+/*
+ <task title>
+ ============
+
+ <task description>
+ [X] checked list item
+ [ ] unchecked list item
+
+ Location: <location>
+ Start: <start date time> <timezone>
+ Due: <due date time> <timezone>
+ Completed: <due date time> <timezone>
+ Priority: <priority text>
+ Privacy: <privacy text>
+ Status: <status text>
+ <url>
+
+ --
+ Shared by OpenTasks
+ */
+
+
+/**
+ * The text description for the whole task to share with other apps. (See the format above.)
+ *
+ * @author Gabor Keszthelyi
+ */
+public class ShareTaskText extends AbstractStringCharSequence implements TaskText
+{
+
+    public ShareTaskText(ContentSet contentSet, Model model, Context context)
+    {
+        super(new Composer(contentSet, model, context).create());
+    }
+
+
+    private static class Composer
```

Comment:
I created it to avoid passing around those parameters to all methods. But I've removed it now.
Just to realize that it may be good as a Factory...  Anyway, added TODO as well to replace with Factory when we have those from the new library.